.github/workflows/build.yaml

```diff
@@ -121,3 +121,55 @@ jobs:
 
       - name: Run phpunit/phpunit
         run: vendor/bin/phpunit
+
+  code-coverage:
+    name: Code Coverage
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version:
+          - 7.2
+
+    steps:
+      - name: Checkout commit
+        if: github.event_name == 'push'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.sha }}
+
+      - name: Checkout head commit
+        if: github.event_name == 'pull_request'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Looks like Scrutinizer accepts the code coverage report (see https://github.com/nikolaposa/version/pull/31/checks?check_run_id=950559775), but cannot attribute it to the commit hash.

`actions/checkout` checks out a theoretical merge commit for pull requests. This commit obviously is not known to Scrutinizer.

By conditionally checking out

- the commit on `push` events (for building pushes to `master`, whether direct or indirect through merges)
- the actual head commit for `pull_request` events

we can work around the issue.

Here, we access the `github` context, which allows inspecting event payloads!

💁‍♂️ For reference, see:

- https://github.com/actions/checkout#Checkout-pull-request-HEAD-commit-instead-of-merge-commit
- https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contexts
- https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context
- https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push
- https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#push
- https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request
- https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#pull_request

Reply:
Wow, that is an amazing find, well done. 👏